### ISSUE TITLE
About EqualLR 

### ISSUE THREAD

**Issue by kaonashi-tyc** [rosinality/style-based-gan-pytorch#75]
Hi,
    Thanks for this awesome reproduction of the StyleGAN paper, it helps me greatly understanding the intrinsics of the model.
    I have one question though about the EqualLR:

https://github.com/rosinality/style-based-gan-pytorch/blob/a3d000e707b70d1a5fc277912dc9d7432d6e6069/model.py#L34-L47

Per my little experiments, this __call__ evaluates every time forward is called. Is this the intended behavior? The original implementation seems to only applies this manipulation at the initialization time. Or per your experience, this doesn't make a difference?

**Comment by rosinality** [comment]
Yes, equalized learning rate requires to apply scaling at every forward. So actually original implementation also applies scaling at each forward time.

**Comment by kaonashi-tyc** [comment]
Thanks for the timely reply.

I found this in the original implementation:

`def get_weight(shape, gain=np.sqrt(2), use_wscale=False, lrmul=1):
    fan_in = np.prod(shape[:-1]) # [kernel, kernel, fmaps_in, fmaps_out] or [in, out]
    he_std = gain / np.sqrt(fan_in) # He init


    # Equalized learning rate and custom learning rate multiplier.
    if use_wscale:
        init_std = 1.0 / lrmul
        runtime_coef = he_std * lrmul
    else:
        init_std = he_std / lrmul
        runtime_coef = lrmul


    # Create variable.
    init = tf.initializers.random_normal(0, init_std)
    return tf.get_variable('weight', shape=shape, initializer=init) * runtime_coef`

Haven't touch TF for a while, so correct me if I am wrong. 

The returned `**tf.get_variable('weight', shape=shape, initializer=init) * runtime_coef**`` is actually an expression and will evaluate every time it gets accessed? And the EqualLR is trying to reproduce the same effects?

**Comment by rosinality** [comment]
Yes, it will apply scaling at each evaluation. You can check Progressive GAN papers https://arxiv.org/abs/1710.10196 for the reasons of it. (contrast to initialization time scaling.)

**Comment by kaonashi-tyc** [comment]
Thanks for the clarification and the pointer. Close the issue :)
